### DISCUSSION TITLE
Support for Java 9 or 10

### DISCUSSION THREAD

**Discussion by Ucifer0931** [javacc/javacc#266] [Ideas]
Hello all, 
          Myself, an newbie on this field. Just an suggestion from my side . I don't know whether @zosrothko is supporting java 9 or 10 version on JavaCC version 8. If it so, then I so happy about that. Or else we should definitely work on that . 
          Products or jar files created using JavaCC as an lot of uses in other products, as time grows many of that products started moving to higher java version (mostly java 11 now ). so it will be more flexible for using.
          Are do we undergoing any difficulties on upgrading to them.If it so you can share about them.

**Comment by zbynek** [comment]
In terms of running JavaCC, you should be fine with Java 11 (tested with Javacc 7, assuming Javacc 8 is the same). In terms of supporting new keywords (yield, record, sealed, ...) in .jj files I don't think there's been any progress, but most of those are just Java 13+ anyway.

**Reply by zosrothko** [reply]
Only @new-javacc , i.e. Sreeni can decide to include or not the new keywords in the JavaCC grammar. Please contact him for a discussion.

**Comment by MarcMazas** [comment]
Well, IMHO upgrading the Java part of the JavaCC grammar to support JDK 9+ syntax should not alter much (= not at all in fact) the core and should be doable by anybody (assuming he provides the test cases).

**Comment by MarcMazas** [comment]
We intend to provide JavaCC grammars for the different Java versions.
We'll have to look at the issues that can arise if we integrate these in the JavaCC grammars. Most probably it would be done only in JavaCC 8 / Java.
I would like to hear of examples of Java 9+ features that would be useful in the core part of a grammar (the actions, the semantic lookaheads and the method headers) and that cannot easily be coded in helper classes (keeping only method calls in the generated parser class).

**Comment by new-javacc** [comment]
I suggest we move away from that part. A better way would be to have a full java 9 (or 17? lol) grammar that you can use to parse the generated code for validation rather than trying to parse it inline so the codegenerator can be clean
